src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1393,8 +1393,8 @@ public ResultSet getTables(
       } else if (schemaPattern.isEmpty()) {
         return SnowflakeDatabaseMetaDataResultSet.getEmptyResultSet(GET_TABLES, statement);
       } else {
-        schemaPattern = unescapeChars(schemaPattern);
-        showCommand += " in schema \"" + catalog + "\".\"" + schemaPattern + "\"";
+        String schemaUnescaped = unescapeChars(schemaPattern);
+        showCommand += " in schema \"" + catalog + "\".\"" + schemaUnescaped + "\"";
```

Comment:
a better solution is to use `final` for all variables needed by telemetry in the function parameter body to guarantee those variable cannot be changed in this method.

Reply:
The catalog and schema can get changed from their original values depending on the values of the session parameters CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX and CLIENT_METADATA_USE_SESSION_DATABASE. I want to record the telemetry values after the catalog and schema have been changed (to reflect what's really being used), but prior to removing the escape values from the parameters. So catalog and schema can't be final, but the other 2 parameters can.

Reply:
Actually catalog and schema can be made final after the values are retrieved from applySessionContext() function as well

Reply:
just make sure we can log the original inputs to telemetry then we can repro the issue offline once we have the telemetry.